### PR TITLE
Remove the trial license key requirement

### PR DESCRIPTION
**Pull request by radazzouz**
# Details

⚠️ Before reviewing this PR, please take a look at https://github.com/PSPDFKit/PSPDFKit-Website/pull/5233 first. ⚠️ 

The new getting started instructions from the [README](https://github.com/PSPDFKit/react-native/blob/rad/remove-trial-license-key-requirement/README.md) are very similar to the ones from https://pspdfkit.com/getting-started/mobile/?frontend=react-native&react-native-platform=android&project=new-project

## Release Notes

- Remove the trial license key requirement.
- Update to PSPDFKit for Android version 6.6.2.
- Update the README to clarify the Requirements section.
- Update the README to simplify the Getting Started section.

# Acceptance Criteria

- [x] Test the Catalog and Native Catalog example projects on Android and iOS.
- [x] Test a newly created project on Android and iOS using the new Getting Started Instructions in the README. When testing, please use `yarn add github:PSPDFKit/react-native#rad/remove-trial-license-key-requirement` for the fourth step.
- [x] When approved, right before merging, rebase with master and increment the package version in `package.json`, `package-lock.json`, `samples/Catalog/package.json`, and `samples/NativeCatalog/package.json` (see example commit:  https://github.com/PSPDFKit/react-native/pull/389/commits/ae027329b5380906fa778354945178c7e92730a4).
- [ ] Create a new release (and tag) with the new package version (see https://github.com/PSPDFKit/react-native/releases).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,117 +37,168 @@ To make sure this is fast, please use a work email and have someone from your co
...
+
+```bash
+open ios/Podfile
+```
+
+Your Podfile should look like this:
 
 ```diff
 require_relative '../node_modules/react-native/scripts/react_native_pods'
+require_relative '../node_modules/react-native/scripts/react_native_pods'
```
Comment (steviki) [inline README.md:89]: This is listed twice.
I guess by accident?

```suggestion
```

File: README.md [hunk 2]
```diff
@@ -37,117 +37,168 @@ To make sure this is fast, please use a work email and have someone from your co
...
+```bash
+react-native run-ios
 ```
 
-Your app is now ready to launch. Run the app in Xcode or go back to the Terminal, then run `cd ..`, and `react-native run-ios`.
-
 ### Usage
 
-There are 2 different ways on how to use the PSPDFKit React Native wrapper on iOS.
+There are 2 different ways on how to use the PSPDFKit for React Native on iOS.
```
Comment (steviki) [inline README.md:201]:
```suggestion
There are 2 different ways on how to use PSPDFKit for React Native on iOS.
```

**Review by steviki — APPROVED** [review]
iOS changes look good to me!

File: README.md (hunk 1 above)
Comment (radazzouz) [reply README.md:89]: Yes. It was an accident 🙈

File: README.md [hunk 3]
```diff
@@ -397,27 +468,18 @@ Let's create a simple app that integrates PSPDFKit and uses the react-native-psp
...
-7. PSPDFKit targets modern platforms, so you'll have to set the `minSdkVersion` to 19. In `YourApp/android/build.gradle`:
+8. Open the app’s `build.gradle` file: 
 
-```diff
-...
- buildscript {
-     ext {
-         buildToolsVersion = "28.0.3"
--        minSdkVersion = 16
-+        minSdkVersion = 19
```
Comment (amit-nayar) [inline README.md:408]: Do we not need to set this anymore?

**Review by amit-nayar — APPROVED** [review]
Android guide LGTM

**Pull request merged**